target-app/Chat/Views/ConversationTableViewCell.swift

```diff
@@ -0,0 +1,110 @@
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.alignment = .leading
+        stackView.distribution = .fill
+        stackView.spacing = 8
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        return stackView
+    }()
+    
+    private var unreadMessagesLabel: UILabel = {
+        let label = UILabel(style: .secondary(textColor: .white))
+        label.layer.cornerRadius = 20 / 2
+        label.layer.masksToBounds = true
+        label.backgroundColor = .systemYellow
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        contentView.addSubviews(subviews: [
+            avatarImage,
+            stackView,
+            topicIcon,
+            unreadMessagesLabel
+        ])
+        
+        stackView.addArrangedSubview(subviews: [
+            fullNameLabel,
+            lastMessageLabel,
+        ])
+        
+        avatarImage.centerVertically(with: contentView)
+        topicIcon.centerVertically(with: contentView)
+        stackView.centerVertically(with: contentView)
+        
+        NSLayoutConstraint.activate([
+            avatarImage.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: avatarImageSpacing),
+            stackView.leadingAnchor.constraint(equalTo: avatarImage.trailingAnchor, constant: stackViewSpacing),
+            stackView.trailingAnchor.constraint(equalTo: topicIcon.leadingAnchor, constant: stackViewSpacing),
+            topicIcon.leadingAnchor.constraint(equalTo: stackView.trailingAnchor, constant: topicIconSpacing),
+            topicIcon.trailingAnchor.constraint(equalTo: contentView.trailingAnchor,constant: -topicIconSpacing),
+            unreadMessagesLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -unreadMessagesSpacing)
+        ])
+
+        NSLayoutConstraint.activate([
+            avatarImage.heightAnchor.constraint(equalToConstant: avatarImageSize),
+            avatarImage.widthAnchor.constraint(equalToConstant: avatarImageSize),
+            topicIcon.topAnchor.constraint(equalTo: topAnchor),
+            topicIcon.bottomAnchor.constraint(equalTo: bottomAnchor),
+            topicIcon.heightAnchor.constraint(equalToConstant: topicIconSize),
+            topicIcon.widthAnchor.constraint(equalToConstant: topicIconSize),
+            unreadMessagesLabel.heightAnchor.constraint(equalToConstant: unreadMessagesLabelSize),
+            unreadMessagesLabel.widthAnchor.constraint(equalToConstant: unreadMessagesLabelSize),
+            unreadMessagesLabel.topAnchor.constraint(equalTo: contentView.topAnchor, constant: unreadMessagesSpacing)
+        ])
+    }
+    
+    func updateData(match: Match) {
+        avatarImage.image = UIImage(named: match.user.avatar.smallThumbUrl!)
```

Comment:
Pls avoid force unwrapping, specially with data like this. You shouldn't trust that the thumbnailURL will be present.
In a different topic, please add helpers to the match model to hide its internals and avoid this level of indirection (match.user.avatar.smallThumbUrl)

Reply:
cool thanks, done!